### ISSUE TITLE
Estimated time should consider respawns

### ISSUE THREAD

**Issue by timvermeulen** [jamesy012/OpenPlanet-BestCheckpoints#21]
I've not personally used this plugin, I've just seen others use it, so apologies if I get some details wrong.

The estimated time starts to increase when the current CP is taking too long compared to whatever the estimated time is set to use. It doesn't seem to take respawns into account, though. If a CP is estimated to take 15 seconds, but you respawn 10 seconds into the CP, it will stay constant for another 5 seconds before starting to increase. This means that if a respawn has happened, you need to mentally add some time (depending on how long the CP is and whether the estimated time is already increasing) to get a more accurate estimate. I argue that if you respawn 10 seconds into the CP, the timer should jump up by 10 seconds immediately, and only start to increase 15 seconds after that (provided no new respawns happen).

Sorry for the poor description, I hope the idea is clear.

**Comment by jamesy012** [comment]
yeah it's something I want to look into at some point, the plugin doesnt really deal with respawns all too well (just ignores them)
I was thinking of adding an option that resets that checkpoints time when a respawn happens
but havent actually done anything yet about it

**Comment by lgarron** [comment]
> I was thinking of adding an option that resets that checkpoints time when a respawn happens
> but havent actually done anything yet about it

I would be really glad to see this!

I was confused for a long time why my sum of bests was sometimes significantly slower than my no-respawn timer for a single run.

**Comment by lgarron** [comment]
I have a draft of this working at https://github.com/jamesy012/OpenPlanet-BestCheckpoints/compare/master...lgarron:OpenPlanet-BestCheckpoints:respawn?expand=1

Unfortunately, it required commenting out some code in `GetPlayerCheckpointTime()` that I think should actually stay.

Since I only play Trackmania 2020, I also can't evaluate or test the other `#ifdef` branches.

**Comment by jamesy012** [comment]
Hey, Thanks for taking the time to look at this.
Do you mind if I integrate it into this version?

I really should get around to updating the plugin as well.

I mostly only play nations/united forever which open planet does
not support...

**Comment by lgarron** [comment]
> Hey, Thanks for taking the time to look at this. Do you mind if I integrate it into this version?

Yeah, please go ahead! I'd be really glad to have the functionality in a signed version.
